### PR TITLE
[SOAR-17028] [Proofpoint TAP] Update connection schema to required:true

### PR DESCRIPTION
**Pull request by joneill-r7**
## Proposed Changes
- [[Proofpoint TAP] Update connection schema to correct fields as required](https://rapid7.atlassian.net/browse/SOAR-17028)

We need to change the fields for the connection to be required as it is allowing customers to create connection objects with no associated credentials which then causes the actions/tasks to fail when they are triggered. 

### Description
Describe the proposed changes:

  - Ran refresh which has re-ordered the help.md and required some fields populated into the plugin spec.
  - Updated connection schema so that the user is required to enter values. One action: parse alerts does not require a connection but due to how we run the action the plugin sidecar still tries to decrypt the connection associated with the plugin and fails on decrypting empty values. 

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [X] ~Unit tests written for any new or updated code~

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/proofpoint_tap/help.md [hunk 1]
```diff
@@ -74,225 +74,165 @@ Example input:
...
-        "threatStatus": "active",
-        "threatTime": "2021-04-20T21:08:38.000Z",
-        "threatURL": "https://example.com",
-        "url": "https://example.com",
-        "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.128 Safari/537.36"
-      }
-    ],
-    "queryEndTime": "2021-04-21T13:00:00Z"
-  }
+  "results": {}
```
Comment (rbowden-r7) [inline plugins/proofpoint_tap/help.md:229]: There looks to be a handful of examples that are not added to the spec and are coming in as empty to help md

Comment (rbowden-r7) [reply plugins/proofpoint_tap/help.md:229]: These include the outputs for the following 

- Get All Threats
- Get Blocked Clicks
- Get Blocked Messages
- Get Delivered Threats
- Get Permitted Clicks
- Get Top Clickers
- Parse Alert
- URL Decode

**Review by jerichardson-r7 — APPROVED** [review]

**Review by igorski-r7 — APPROVED** [review]

File: plugins/proofpoint_tap/help.md [hunk 2]
```diff
@@ -49,9 +47,11 @@ Example input:
 
 ### Actions
 
+
 #### Fetch Forensics
 
-This action is used to fetch forensic evidence about individual threats or campaigns. Either 'threatId' or 'campaignId' should be specified.
+This action is used to fetch Forensics allows users to pull detailed forensic evidence about individual threats or
```
Comment (ekelly-r7) [inline plugins/proofpoint_tap/help.md:53]: small typo in this sentence

**Review by jerichardson-r7 — APPROVED** [review]

File: plugins/proofpoint_tap/help.md [hunk 3]
```diff
@@ -434,743 +432,751 @@ Example input:
...
+      }
+    }
+  ]
 }
 ```
 
 #### Parse Alert
 
-This action is used to parse a TAP alert. It is often used to parse the indicators from a forwarded e-mail containing a TAP alert. Note that this action does not require the connection to be configured.
+This action is used to parses a TAP alert
```
Comment (ekelly-r7) [inline plugins/proofpoint_tap/help.md:660]: small typo `parses` should be `parse`

File: plugins/proofpoint_tap/help.md (hunk 2 above)
Comment (joneill-r7) [reply plugins/proofpoint_tap/help.md]: Updated this now :)

File: plugins/proofpoint_tap/help.md (hunk 3 above)
Comment (joneill-r7) [reply plugins/proofpoint_tap/help.md]: Updated also

**Review by ekelly-r7 — APPROVED** [review]

**Review by rbowden-r7 — APPROVED** [review]

**Pull request merged**